utils/initial_data_preparation.py

```diff
@@ -31,52 +83,58 @@ def check_and_clean_data(df):
-            # If there are three columns, append them directly
-            a_values.append(columns[0])
-            b_values.append(columns[1])
-            c_values.append(columns[2])
-        elif len(columns) == 1:
-            # If there is one column, use the previous 'a' and 'b' values, and
-            # append 'c'
-            a_values.append(a_values[-1] if a_values else pd.NA)
-            b_values.append(b_values[-1] if b_values else pd.NA)
-            c_values.append(columns[0])
-        else:
-            raise ValueError(f"Unexpected number of columns: {len(columns)}")
-
-
-# Create a DataFrame from the accumulated lists
-df = pd.DataFrame({"projectref": a_values, "nlnetpage": b_values, "repourl": c_values})
-
-
-# Save the dataframe as a CSV file
-path_to_data_folder = git_working_dir / "data"
-
-# Applying preprocessing steps
-check_and_clean_data(df)
-
-# I will keep the first occurrence of each duplicate row and remove the
-# others:
-df = df.drop_duplicates(keep="first")
-df.to_csv(path_to_data_folder / "original.csv", index=False)
-
-# Extracting the rows which host the code on github.com domain
-original_github_df = df[df["repourl"].str.contains("github.com")]
-original_github_df.to_csv(path_to_data_folder / "original_github_df.csv", index=False)
+if __name__ == "__main__":
+    args = parse_args()
+
+    git_working_dir = get_working_directory_or_git_root()
+    logger.info(f"git_working_dir is : {git_working_dir}")
+
+    # Define the full input path
+    df_path = Path(args.input_file)
+    logger.info(f"Input file path is : {df_path}")
+
+    # Define the directory to save output files and ensuring the directory
+    # exists
+    output_dir = Path(args.output_folder)
+    logger.info(f"Output file path is: {output_dir}")
+    output_dir.mkdir(parents=True, exist_ok=True)
+
+    # Initialize lists to hold the column data
+    a_values, b_values, c_values = [], [], []
+
+    # Read the file line by line
+    with open(df_path, "r") as file:
+        for line in file:
+            columns = line.strip().split("\t")
+
+            if len(columns) == 3:
+                # If there are three columns, append them directly
+                a_values.append(columns[0])
+                b_values.append(columns[1])
+                c_values.append(columns[2])
+            elif len(columns) == 1:
+                # If there is one column, use the previous 'a' and 'b' values, and
+                # append 'c'
+                a_values.append(a_values[-1] if a_values else pd.NA)
+                b_values.append(b_values[-1] if b_values else pd.NA)
+                c_values.append(columns[0])
+            else:
+                raise ValueError(f"Unexpected number of columns: {len(columns)}")
+
+    # Create a DataFrame from the accumulated lists
+    df = pd.DataFrame(
+        {"projectref": a_values, "nlnetpage": b_values, "repourl": c_values}
+    )
+
+    # Applying preprocessing steps
+    check_and_clean_data(df)
```

Comment:
Does `check_and_clean_data(...)` perform the operations in-situ? in other words does it update the contents of the dataframe? For consistency (and this can be in a separate PR) I'd recommend we structure the function so it works more like df.drop_duplicates where it returns an updated dataframe which the caller then assigns to whichever dataframe variable they choose (as per line 133 here: `df = df.drop_duplicates(keep="first")`).